CoordImporter/Parser/Parsers.cs

```diff
@@ -1,113 +1,137 @@
+        RegexOptions.Compiled);
+
 
-    public Result<MarkData, string> Parse(string inputLine)
+    public override Result<MarkData, string> Parse(string inputLine)
     {
-        var groups = SirenRegex().Matches(inputLine).First().Groups;
-        ITrackerParser.LogParse("Siren", inputLine, groups);
-        return groups.CreateMark(instance => (uint)(instance.First() - I1Char) + 1);
+        var groups = Regex.Matches(inputLine).First().Groups;
+        LogParse("Siren", inputLine, groups);
+        return CreateMark(groups, instance => (uint)(instance.First() - I1Char) + 1);
     }
+
+    public override bool CanParseLine(string inputLine) => Regex.IsMatch(inputLine);
+
+    public SirenParser(IPluginLog logger, IDataManagerManager dataManagerManager) : base(logger, dataManagerManager) { }
 }
 
-public partial class FaloopParser : ITrackerParser
+public class FaloopParser : Parsers
 {
     // For the format "Raiden [S]: Gamma - Yanxia ( 23.6, 11.4 )"
-    [GeneratedRegex(@"(?<world_name>[a-zA-Z0-9'-]+)\s+\[S\]: (?<mark_name>[\w '-]+) - (?<map_name>[\w '-]+)\s+\(?(?<instance>[1-9]?)\)?\s*\(\s*(?<x_coord>[0-9\.]+)\s*,\s*(?<y_coord>[0-9\.]+)\s*\)")]
-    private static partial Regex FaloopRegex();
+    protected readonly Regex Regex = new Regex(
+        @"(?<world_name>[a-zA-Z0-9'-]+)\s+\[S\]: (?<mark_name>[\w '-]+) - (?<map_name>[\w '-]+)\s+\(?(?<instance>[1-9]?)\)?\s*\(\s*(?<x_coord>[0-9\.]+)\s*,\s*(?<y_coord>[0-9\.]+)\s*\)",
+        RegexOptions.Compiled);
 
-    public bool CanParseLine(string inputLine) => FaloopRegex().IsMatch(inputLine);
 
-    public Result<MarkData, string> Parse(string inputLine)
+    public override Result<MarkData, string> Parse(string inputLine)
     {
-        var groups = FaloopRegex().Matches(inputLine).First().Groups;
-        ITrackerParser.LogParse("Faloop", inputLine, groups);
-        return groups.CreateMark(instance => (uint)(instance.First() - '0'));
+        var groups = Regex.Matches(inputLine).First().Groups;
+        LogParse("Faloop", inputLine, groups);
+        return CreateMark(groups, instance => (uint)(instance.First() - '0'));
     }
+
+    public override bool CanParseLine(string inputLine) => Regex.IsMatch(inputLine);
+
+    public FaloopParser(IPluginLog logger, IDataManagerManager dataManagerManager) : base(logger, dataManagerManager) { }
 }
 
-public partial class BearParser : ITrackerParser
+public class BearParser : Parsers
 {
     // For the format "Labyrinthos ( 16.5 , 16.8 ) Storsie"
-    [GeneratedRegex(@"(?<map_name>[\D'\s-]+)\s*(?<instance>[1-9]?)\s*\(\s*(?<x_coord>[0-9\.]+)\s*,\s*(?<y_coord>[0-9\.]+)\s*\)\s*(?<mark_name>\w[\w -]+)")]
-    private static partial Regex BearRegex();
+    protected readonly Regex Regex = new Regex(
+        @"(?<map_name>[\\D'\\s-]+)\\s*(?<instance>[1-9]?)\\s*\\(\\s*(?<x_coord>[0-9\\.]+)\\s*,\\s*(?<y_coord>[0-9\\.]+)\\s*\\)\\s*(?<mark_name>\\w[\\w -]+)",
```

Comment:
XD i ran into this, too. notice the double backslashes. you can fix that by taking the `@` off, then using the Rider context actions to convert it to a verbatim string.